src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/status/StatusBarWidget.java

```diff
@@ -105,15 +114,51 @@ else if (type == StatusBar.SCOPE_SLIDE)
       else
          scopeIcon_.setResource(CodeIcons.INSTANCE.function());
    }
+   
+   public void showMessage(String message)
+   {
+      scope_.setVisible(false);
+      panel_.setCellWidth(scope_, "0");
+      
+      message_.setVisible(true);
+      panel_.setCellWidth(message_, "100%");
+      
+      message_.setValue(message);
+      if (handler_ != null)
+      {
+         handler_.removeHandler();
+         handler_ = null;
+      }
+      
+      handler_ = Event.addNativePreviewHandler(new NativePreviewHandler()
+      {
+         
+         @Override
+         public void onPreviewNativeEvent(NativePreviewEvent event)
```

Comment:
Good point. There are a couple of other options here:

1. Take over `Enter` and have it 'end' the rename (without inserting a newline),
2. End on mouse clicks,
3. End if focus lost.

I'll add these, but can you think of any other candidates?